File: engine/src/main/java/com/bartlomiejpluta/base/engine/world/icon/asset/IconSetAsset.java
package com.bartlomiejpluta.base.engine.world.icon.asset;

import com.bartlomiejpluta.base.engine.common.asset.Asset;
import lombok.Getter;
import lombok.NonNull;

@Getter
public class IconSetAsset extends Asset {
   private final int rows;
   private final int columns;

   public IconSetAsset(@NonNull String uid, @NonNull String source, int rows, int columns) {
      super(uid, source);
      this.rows = rows;
      this.columns = columns;
   }
}

File: editor/src/main/kotlin/com/bartlomiejpluta/base/editor/iconset/asset/IconSetAsset.kt
package com.bartlomiejpluta.base.editor.iconset.asset

import com.bartlomiejpluta.base.editor.asset.model.Asset
import com.bartlomiejpluta.base.editor.asset.model.GraphicAsset
import com.bartlomiejpluta.base.editor.project.model.Project

class IconSetAsset(project: Project, uid: String, source: String, name: String, rows: Int, columns: Int) :
   GraphicAsset(project.iconSetsDirectoryProperty, uid, source, name, rows, columns)